scylla-cdc-printer/src/printer.rs

```diff
@@ -1,220 +1,42 @@
-use futures::future::RemoteHandle;
-use futures::stream::{FusedStream, FuturesUnordered, StreamExt};
-use futures::FutureExt;
-use scylla::frame::response::result::Row;
-use scylla::Session;
-
-use scylla_cdc::cdc_types::GenerationTimestamp;
-use scylla_cdc::reader::StreamReader;
-use scylla_cdc::stream_generations::GenerationFetcher;
-
-pub struct CDCLogPrinter {
-    // Tells the worker to stop
-    // Usage of the "watch" channel will make it possible to change the the timestamp later,
-    // for example if somebody loses patience and wants to stop now not later
-    end_timestamp: tokio::sync::watch::Sender<Duration>,
-}
-
-impl CDCLogPrinter {
-    // Creates a printer and runs it immediately
-    #[allow(clippy::too_many_arguments)]
-    pub fn new(
-        session: &Arc<Session>,
-        keyspace: String,
-        table_name: String,
-        start_timestamp: Duration,
-        end_timestamp: Duration,
-        window_size: Duration,
-        safety_interval: Duration,
-        sleep_interval: time::Duration,
-    ) -> (Self, RemoteHandle<anyhow::Result<()>>) {
-        let (end_timestamp_sender, end_timestamp_receiver) =
-            tokio::sync::watch::channel(end_timestamp);
-        let mut worker = CDCLogPrinterWorker::new(
-            session,
-            keyspace,
-            table_name,
-            start_timestamp,
-            end_timestamp,
-            window_size,
-            safety_interval,
-            sleep_interval,
-            end_timestamp_receiver,
+use async_trait::async_trait;
+use scylla_cdc::consumer::{CDCRow, Consumer, ConsumerFactory};
+
+struct PrinterConsumer {}
+
+#[async_trait]
+impl Consumer for PrinterConsumer {
+    async fn consume_cdc(&mut self, data: CDCRow<'_>) -> anyhow::Result<()> {
+        // [TODO]: Add prettier printing
+        println!(
+            "time: {}, batch_seq_no: {}, end_of_batch: {}, operation: {:?}, ttl: {:?}",
+            data.time, data.batch_seq_no, data.end_of_batch, data.operation, data.ttl
         );
-        let (fut, handle) = async move { worker.run().await }.remote_handle();
-        tokio::task::spawn(fut);
-        let printer = CDCLogPrinter {
-            end_timestamp: end_timestamp_sender,
-        };
-        (printer, handle)
-    }
-
-    // Tell the worker to set the end timestamp and then stop
-    pub fn stop_at(&mut self, when: Duration) {
-        let _ = self.end_timestamp.send(when);
-    }
-
-    // Tell the worker to stop immediately
-    pub fn stop(&mut self) {
-        self.stop_at(Duration::min_value());
+        Ok(())
     }
 }
 
-struct CDCLogPrinterWorker {
-    session: Arc<Session>,
-    keyspace: String,
-    table_name: String,
-    start_timestamp: Duration,
-    end_timestamp: Duration,
-    sleep_interval: time::Duration,
-    window_size: Duration,
-    safety_interval: Duration,
-    readers: Vec<Arc<StreamReader>>,
-    end_timestamp_receiver: tokio::sync::watch::Receiver<Duration>,
-}
+struct PrinterConsumerFactory {}
```

Comment:
Nit: you can avoid the brackets after the struct name if it has no fields. This is called a "unit" struct. IMO it looks a bit nicer when you don't have to include the brackets every time you instantiate such struct.